sklearn/ensemble/_gb.py

```diff
@@ -344,6 +346,10 @@ def _clear_state(self):
             del self.train_score_
```

Comment:
This is for my education only, unrelated to this PR: why are all these `del` calls needed? `_clear_state` is called from `fit()`, so naively I'd expect that the attributes get updated/overwritten during the new fit, so no need to clear them explicitly. Does anyone know?

Reply:
This is indeed a good question. I should check what happens if we don't clear :)

Reply:
So here, the newly added test will fail because we internally detect if we are in a `warm_start` config if `estimators_` is not empty. So if you don't clear the state, it will always believe you are warm-starting.

We might find a more explicit way than how it is done now, I did not look at it.

Reply:
Thanks for the explanation. Something to ponder :)